lib/AST/Decl.cpp

```diff
@@ -1031,7 +1031,7 @@ unsigned PatternBindingDecl::getPatternEntryIndexForVarDecl(const VarDecl *VD) c
     ++Result;
   }
   
-  assert(0 && "PatternBindingDecl doesn't bind the specified VarDecl!");
+  llvm_unreachable("PatternBindingDecl doesn't bind the specified VarDecl!");
```

Comment:
You should check with individual code owners whether the recovery is intended for release builds. If so, `assert(0)` is correct.

(We discourage this in general, but for parsing in particular handling unexpected scenarios has a huge benefit to the user, since it keeps SourceKit from crashing.)